tools/json_schema/json_schema-writers-outputs.adb

```diff
@@ -0,0 +1,650 @@
+-- Software  Foundation;  either version 3,  or (at your  option) any later --
+-- version. This library is distributed in the hope that it will be useful, --
+-- but WITHOUT ANY WARRANTY;  without even the implied warranty of MERCHAN- --
+-- TABILITY or FITNESS FOR A PARTICULAR PURPOSE.                            --
+--                                                                          --
+-- As a special exception under Section 7 of GPL version 3, you are granted --
+-- additional permissions described in the GCC Runtime Library Exception,   --
+-- version 3.1, as published by the Free Software Foundation.               --
+--                                                                          --
+-- You should have received a copy of the GNU General Public License and    --
+-- a copy of the GCC Runtime Library Exception along with this program;     --
+-- see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see    --
+-- <http://www.gnu.org/licenses/>.                                          --
+--                                                                          --
+------------------------------------------------------------------------------
+
+package body JSON_Schema.Writers.Outputs is
+
+   procedure Write_Enum_Specification
+     (Name     : VSS.Strings.Virtual_String;
+      Property : VSS.Strings.Virtual_String;
+      Schema   : Schema_Access;
+      Optional : Boolean);
+   --  Generate Output procedure specification for an enumeration schema
+
+   procedure Write_Enum_Body
+     (Name     : VSS.Strings.Virtual_String;
+      Property : VSS.Strings.Virtual_String;
+      Schema   : Schema_Access;
+      Optional : Boolean);
+   --  Generate Output procedure body for an enumeration schema
+
+   procedure Write_Named_Type
+     (Map    : JSON_Schema.Readers.Schema_Map;
+      Name   : VSS.Strings.Virtual_String;
+      Schema : Schema_Access;
+      Kind   : Declaration_Kind);
+
+   procedure Write_Anonymous_Type
+     (Enclosing_Type : VSS.Strings.Virtual_String;
+      Property       : JSON_Schema.Property;
+      Map            : JSON_Schema.Readers.Schema_Map);
+
+   procedure Write_Output_Specification
+     (Type_Name : VSS.Strings.Virtual_String;
+      Prefix    : VSS.Strings.Virtual_String);
+
+   procedure Write_Record_Component
+     (Name     : VSS.Strings.Virtual_String;
+      Map      : JSON_Schema.Readers.Schema_Map;
+      Property : JSON_Schema.Property;
+      Required : Boolean);
+
+   function Drop_Last_Word
+     (Text : VSS.Strings.Virtual_String)
+      return VSS.Strings.Virtual_String;
+
+   --------------------
+   -- Drop_Last_Word --
+   --------------------
+
+   function Drop_Last_Word
+     (Text : VSS.Strings.Virtual_String) return VSS.Strings.Virtual_String
+   is
+      List : constant VSS.String_Vectors.Virtual_String_Vector :=
+        Text.Split ('_');
+      Result : VSS.Strings.Virtual_String := List (1);
```

Comment:
Should Virtual_String_Vector.Join operation be added to do this? Could you open ticket to implement it if necessary?

Reply:
It would be nice to have `Join`, but for now I've rewritten code to avoid this.